### ISSUE TITLE
change inconsistent API: close, release, terminate

### ISSUE THREAD

**Issue by derickbailey** [oracle/node-oracledb#338]
I would like to suggest changing any method that deals with the closing/releasing/finalizing/terminating/end-of-life-ing/etc-ing of an object, to a single, consistent method name. I don't particularly care what the method name is, as long as it is consistent across all objects. 

At this point, having to remember if I am supposed to `close` or `terminate` or `release` an object when i'm done with it, is causing me headaches as a developer. I can never remember which method to use, aon which object (connection, cursor, pool) and I constantly have to either look it up or guess and watch my code crash until i get it right.

**Comment by cjbj** [comment]
@derickbailey Good idea.  In fact it's already on the long todo list to review.  I'd noted that we should add resultset.release() and document that resultset.close() is obsolete.  I guess Pool.terminate() should likewise be Pool.release()?

Perhaps this is something that can be easily added to @dmcghan's https://github.com/oracle/node-oracledb/pull/321#issuecomment-173316224

**Comment by sagiegurari** [comment]
ya, once you can customize in the js layer the objects, you can simply do:
````js
pool.release = pool.terminate;
resultset.release = resultset.close;
````

**Comment by cjbj** [comment]
While the issue's hot, yell out if anyone prefers a word other than 'release'

**Comment by derickbailey** [comment]
i think i prefer `.close()` ... or if i reach back to my .NET days with the IDisposable interface, `.dispose()`. but again, I'm not super picky about what the method name is, as long as it's consistent across the objects.

being able to modify the object in JS will be great, @sagiegurari! that will certainly help make it easier to contribute and to put in temporary workarounds like that :)

**Comment by dmcghan** [comment]
Sorry for being late to the party. To summarize, the suggestions for a consistent method are:
- close
- terminate
- release
- dispose

Here are the relevant classes in which the method would apply:
- pool
- connection
- lob
- resultset
- stream

I'm not a fan of terminate (or destroy). I've vote for either: `close`, `release`, or `dispose`.

**Comment by dmcghan** [comment]
I put out a Twitter poll to (hopefully) gather some opinions:
https://twitter.com/dmcghan/status/710852254568484864

**Comment by sagiegurari** [comment]
In simple-oracledb I added close to connection and pool so I'm definitely for that.
Resultset already has close.

**Comment by soyuka** [comment]
Some thoughts about this.

IMO, on one hand it makes sense to "release" a connection if it's in a pool. Obviously, the connection closes, but it's giving *space* for some more connections, hence the "release" word (free). 
On the other hand, "the pool is closed" means that you won't be able to add more connections. 

I answered "close" on the Twitter poll, but I'm not fond of this word when we talk about a connection pool.

"terminate" is really strong, maybe too strong. It gives the feeling that this is the end, and your program will have to exit. `close` would be just fine :).

About the stream, in nodejs there is an `end` method on writable streams. I didn't use streams from this library but when it comes to method names, I'd try to keep them close to the nodejs ones.

**Comment by dmcghan** [comment]
@soyuka Thanks for your input!

**Comment by dmcghan** [comment]
@soyuka After rereading your comments I'm a little confused...

First you said:
> I answered "close" on the Twitter poll, but I'm not fond of this word when we talk about a connection pool.

And then: 
> close would be just fine

So you like `close` for connection pools?

**Comment by soyuka** [comment]
Indeed, my mistake (edited).

I'd keep `.release` for the connection, but put `.close` instead of `.terminate` for the pool.

**Comment by dmcghan** [comment]
And the results are in:

- 50% close
- 18% terminate
- 18% release
- 14% dispose

Thanks to everyone who voted!

**Comment by dmcghan** [comment]
I’d say the best way to add this "feature" would be via aliases (similar to what lodash and other libraries do). We’d keep the current method names, but add a “close” alias for the following:

- connection.release
- pool.terminate

Lob, ResultSet, and ResultSetReadStream either already have a close method or we're planning on adding one. 

Any thoughts?

**Comment by sagiegurari** [comment]
Agree, that's what I did as well.
That's the only way to be backward compatible.

https://github.com/sagiegurari/simple-oracledb/blob/a3a13c9731e593cc0b76a4165a498b219f2b1590/lib/pool.js#L270-L278

https://github.com/sagiegurari/simple-oracledb/blob/a3a13c9731e593cc0b76a4165a498b219f2b1590/lib/connection.js#L443-L456

**Comment by cjbj** [comment]
This is fixed in node-oracledb 1.9.1.

**Comment by soyuka** [comment]
Could you ref the commit @cjbj? thx
